### PR TITLE
Align log format according to ADR6

### PR DESCRIPTION
**Pull request by mmorhun**
This PR makes Image Controller operator logs aligned with [ADR6](https://github.com/redhat-appstudio/book/blob/main/ADR/0006-log-conventions.md).
Now, by default, Image Controller prints logs in JSON format. Logs example in `json` format:

<details>
<summary>Show log in JSON format</summary>

```
{"level":"info","ts":"2023-04-27T08:23:24.959Z","logger":"setup","caller":"rest/request.go:538","msg":"Waited for 1.037392686s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/k8s.cni.cncf.io/v1?timeout=32s\n"}
{"level":"info","ts":"2023-04-27T08:23:27.168Z","logger":"controller-runtime.metrics","caller":"metrics/listener.go:44","msg":"Metrics server is starting to listen","addr":"127.0.0.1:8080"}
{"level":"info","ts":"2023-04-27T08:23:27.168Z","logger":"setup","caller":"runtime/proc.go:250","msg":"starting manager"}
{"level":"info","ts":"2023-04-27T08:23:27.169Z","caller":"runtime/asm_amd64.s:1571","msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"info","ts":"2023-04-27T08:23:27.169Z","caller":"runtime/asm_amd64.s:1571","msg":"Starting server","path":"/metrics","kind":"metrics","addr":"127.0.0.1:8080"}
{"level":"info","ts":"2023-04-27T08:23:27.169Z","logger":"setup","caller":"leaderelection/leaderelection.go:206","msg":"attempting to acquire leader lease image-controller/ed4c18c3.appstudio.redhat.com...\n"}
{"level":"info","ts":"2023-04-27T08:23:44.677Z","logger":"setup","caller":"wait/wait.go:155","msg":"successfully acquired lease image-controller/ed4c18c3.appstudio.redhat.com\n"}
{"level":"info","ts":"2023-04-27T08:23:44.677Z","caller":"controller/controller.go:241","msg":"Starting EventSource","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","source":"kind source: *v1alpha1.Component"}
{"level":"info","ts":"2023-04-27T08:23:44.677Z","caller":"controller/controller.go:241","msg":"Starting Controller","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component"}
{"level":"info","ts":"2023-04-27T08:23:44.779Z","caller":"controller/controller.go:241","msg":"Starting workers","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","worker count":1}
{"level":"info","ts":"2023-04-27T08:24:23.729Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Waiting for devfile model in component","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"11ab41f5-98cb-47b6-9473-66c2bb49c0fb"}
{"level":"info","ts":"2023-04-27T08:24:23.747Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Waiting for devfile model in component","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"d2cb01bc-a5f4-4fb7-b399-f743a18b219e"}
{"level":"info","ts":"2023-04-27T08:24:35.419Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Created image registry secret go-component-sample for Component","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"c6dfada3-7d0a-4e7f-a8c2-da18fafc90e5","action":"ADD"}
{"level":"info","ts":"2023-04-27T08:24:35.434Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Image regipository finaliziler added to the Component","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"c6dfada3-7d0a-4e7f-a8c2-da18fafc90e5","action":"UPDATE"}
{"level":"info","ts":"2023-04-27T08:24:35.434Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Component updated successfully","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"c6dfada3-7d0a-4e7f-a8c2-da18fafc90e5","action":"UPDATE"}
{"level":"info","ts":"2023-04-27T08:24:49.504Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Deleted robot account testgo-application-samplego-component-sample","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"be7149b8-7f7e-4f66-bad2-30da89029b37","action":"DELETE"}
{"level":"info","ts":"2023-04-27T08:24:50.171Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Deleted image repository test/go-application-sample/go-component-sample","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"be7149b8-7f7e-4f66-bad2-30da89029b37","action":"DELETE"}
{"level":"info","ts":"2023-04-27T08:24:50.193Z","logger":"ComponentImageRepository","caller":"controller/controller.go:121","msg":"Image repository finalizer removed from the Component","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","component":{"name":"go-component-sample","namespace":"test"},"namespace":"test","name":"go-component-sample","reconcileID":"be7149b8-7f7e-4f66-bad2-30da89029b37","action":"DELETE"}
```

</details>

### CODE REVIEW

File: pkg/quay/quay.go [hunk 1]
```diff
@@ -57,47 +58,41 @@ func NewQuayClient(c *http.Client, authToken, url string) QuayClient {
...
-		return nil, err
+		return nil, fmt.Errorf("failed to read response body: %w", err)
 	}
 
 	data := &Repository{}
 	err = json.Unmarshal(body, data)
 	if err != nil {
-		fmt.Println(err)
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshall response body: %w", err)
```
Comment (tnevrlka) [inline pkg/quay/quay.go:88]:
```suggestion
		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
```

File: pkg/quay/quay.go [hunk 2]
```diff
@@ -127,19 +122,52 @@ func (c *QuayClient) DeleteRepository(organization, imageRepository string) (boo
...
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to read response body: %w", err)
 	}
 	data := &QuayError{}
 	err = json.Unmarshal(body, data)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to unmarshall response body: %w", err)
```
Comment (tnevrlka) [inline pkg/quay/quay.go:130]:
```suggestion
		return false, fmt.Errorf("failed to unmarshal response body: %w", err)
```

File: pkg/quay/quay.go [hunk 3]
```diff
@@ -127,19 +122,52 @@ func (c *QuayClient) DeleteRepository(organization, imageRepository string) (boo
...
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	data := &RobotAccount{}
+	err = json.Unmarshal(body, data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall response body: %w", err)
```
Comment (tnevrlka) [inline pkg/quay/quay.go:162]:
```suggestion
		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
```

File: pkg/quay/quay.go [hunk 4]
```diff
@@ -148,68 +176,38 @@ func (c *QuayClient) CreateRobotAccount(organization string, robotName string) (
...
-		return nil, err
+		return nil, fmt.Errorf("failed to read response body: %w", err)
 	}
 
 	data := &RobotAccount{}
 	err = json.Unmarshal(body, data)
 	if err != nil {
-		fmt.Println(err)
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshall response body: %w", err)
```
Comment (tnevrlka) [inline pkg/quay/quay.go:202]:
```suggestion
		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
```

File: pkg/quay/quay.go [hunk 5]
```diff
@@ -243,67 +241,72 @@ func (c *QuayClient) DeleteRobotAccount(organization string, robotName string) (
...
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to read response body: %w", err)
 	}
 	data := &QuayError{}
 	err = json.Unmarshal(body, data)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to unmarshall response body: %w", err)
```
Comment (tnevrlka) [inline pkg/quay/quay.go:249]:
```suggestion
		return false, fmt.Errorf("failed to unmarshal response body: %w", err)
```

**Review by tnevrlka — APPROVED** [review]
absolutely tiny nitpick, thanks for noticing mistakes in my quay functions, LGTM

File: pkg/quay/quay.go (hunk 1 above)
Comment (mmorhun) [reply pkg/quay/quay.go:88]: Thank you, will fix

**Pull request merged**